### PR TITLE
Parse URLs relative to the relevant settings object, not entry

### LINKED ISSUE #922: Eliminate usage of entry and incumbent settings objects
See https://readable-email.org/list/public-script-coord/topic/multiple-globals-and-you. We'd like to stop using these in new specs. In addition, it seems to be the case that most places that use them, are not actually implemented that way in browsers.

It would be good to do an audit of this spec and see which globals are actually implemented in browsers, then update the spec to use those. In the rare cases where incumbent and entry are actually implemented as specced, it would be ideal to change implementations if possible.

For an example of how you can test implementations, see e.g. https://github.com/whatwg/html/pull/1472 and https://github.com/whatwg/html/pull/1473

### PR DESCRIPTION
**Pull request by domenic**
Fixes #922. Web platform tests at https://github.com/w3c/web-platform-tests/pull/3449. Part of the overall effort in https://github.com/whatwg/html/issues/1431.

In the process, fixes some ambiguity about how to parse URLs in the Link header, as previous it parsed relative to the entry settings object, which did not exist during this algorithm.

### CODE REVIEW

File: spec/service_worker/index.bs [hunk 1]
```diff
@@ -2125,29 +2123,31 @@ spec: webidl; urlPrefix: https://heycam.github.io/webidl/
       <li>Let <var>request</var> be the <a for="fetch">request</a> for which this header was received in the response.</li>
       <li>If <var>request</var>'s <a for="request">client</a> is not a <a>secure context</a>, abort these steps.</li>
       <li>Let <var>scriptURL</var> be the result of <a for="url">parsing</a> the <a>target IRI</a> of the <code>Link</code> header.</li>
-      <li>Let <var>scopeURL</var> be the "<code>scope</code>" <a>target attribute</a> of the <code>Link</code> header, or null if no such attribute is present.</li>
+      <li>Let <var>scopeURL</var> null.
```
Comment (jungkees) [inline spec/service_worker/index.bs]: "be" is missing.

File: spec/service_worker/index.bs [hunk 2]
```diff
@@ -2125,29 +2123,31 @@ spec: webidl; urlPrefix: https://heycam.github.io/webidl/
...
       <li>Let <var>promise</var> be a new <a>promise</a>.</li>
       <li>Invoke [[#start-register-algorithm]] with <var>scopeURL</var>, <var>scriptURL</var>, <var>promise</var>, <var>client</var>, <var>client</var>'s <a>creation URL</a> and <var>workerType</var>.</li>
       <li>Run the following substeps <a>in parallel</a>:
         <ol>
           <li>Wait until <var>promise</var> settles.</li>
-          <li>If <var>promise</var> rejected, <a>queue a task</a> to <a>fire a simple event</a> named <code>error</code> at the <{link}> element.</li>
-          <li>If <var>promise</var> resolved, <a>queue a task</a> to <a>fire a simple event</a> named <code>load</code> at the <{link}> element.</li>
+          <li>If <var>promise</var> rejects, <a>queue a task</a> to <a>fire a simple event</a> named <code>error</code> at the <{link}> element.</li>
+          <li>If <var>promise</var> fulfills, <a>queue a task</a> to <a>fire a simple event</a> named <code>load</code> at the <{link}> element.</li>
```
Comment (jungkees) [inline spec/service_worker/index.bs]: "resolves" is wrong here? We used "resolve" in other places.

**Comment by jungkees** [comment]
@domenic, thanks for sorting out the behavior and working on this PR!

LGTM % the above nits

File: spec/service_worker/index.bs (hunk 2 above)
Comment (domenic) [inline spec/service_worker/index.bs]: Yeah, resolves is wrong. https://github.com/domenic/promises-unwrapping/blob/master/docs/states-and-fates.md

Comment (domenic) [inline spec/service_worker/index.bs]: I will revert this and let you do a separate cleanup if you want

Comment (jungkees) [inline spec/service_worker/index.bs]: I will do.

**Pull request merged**